modules/sso/apply/patches/apply-eap-rh-sso-one-off-patches.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e
+
+readonly SOURCES_DIR="/tmp/artifacts/"
+export JAVA_OPTS="${JAVA_OPTS} -Dorg.wildfly.patching.jar.invalidation=true"
+
+# Do not use for cycle, it would faile if no such files are found
+find "${SOURCES_DIR}" \( -name 'eap-one-off-*.zip' -o -name 'rh-sso-*.zip' \) | while read -r I; do
+    echo "Applying patch: '$I' ..."
+    "${JBOSS_HOME}"/bin/jboss-cli.sh --command="patch apply $I"
+done
```

Comment:
This pattern for a loop (pipe to `while`) is robust. It's also not sensitive to whitespace in the values either. No changes needed!

Just for future reference, the pipe format can result in weird bugs in some circumstances, since when you use a pipe, it all becomes one command, and variables set inside the loop will not persist after it is complete. I now prefer the subshell format over pipe format, which avoids the possibility of this bug:

```
while read -r path
do
    echo "$path"
done < <( find "somewhere/" )
```

Reply:
@ASzc Thanks for the review. Right, the only thing I added to the previous version was the `-r` option to the `read` command, as `shellcheck` was complaining without it there can be newlines.

So should the current form be rewritten into the form you suggest below? Is it also able to handle cases when searching for multiple names? 

Afaict original @drichtarik 's form didn't use the escaped parenthesis across / around by `or` composited multiple `-name` argumets -- IOW it used just the `find "${SOURCES_DIR}" -name 'eap-one-off-*.zip' -o -name 'rh-sso-*.zip'`and it seems to have found both of the names too. Just saw the escaped parentheses as the recommended one (maybe on StackOverflow or somewhere), so used that one.

Thanks,
Jan